File: Volume 10 (1000 - 1099)/1046/1046.c
#include <stdio.h>
 
int main() {
    int hi, hf, total = 0;
    scanf("%d %d", &hi, &hf);
    if (hi > hf) hf += 24;
    if (hi == hf) total = 24;
    else total = hf - hi;
    printf("O JOGO DUROU %d HORA(S)\n", total);
    return 0;
}

File: Volume 10 (1000 - 1099)/1046/1046.rb
#!/usr/bin/ruby

a,b = Array.new(gets.split.map(&:to_i))
if b > a
    total = b - a
elsif b == a
    total = 24
else
    total = (b+24) - a
end
puts "O JOGO DUROU %d HORA(S)" % [total]